src/runJscodeshiftCodemod.ts

```diff
@@ -72,6 +74,27 @@ export const runJscodeshiftCodemod = (
 		return commands;
 	}
 
+	// sometimes codemods produce newData even though they are literally no changes
+	// by removing parentheses around return statements, we will likely find the pointless results
+	try {
+		const oldRoot = api.jscodeshift(oldData);
+		const newRoot = api.jscodeshift(newData);
```

Comment:
what will happen if user runs codemod that intentionally removes parentheses? will it show that there is no changes?

Reply:
even if they remove parentheses, recast might added them again (when you run `root.toSource()`). https://github.com/benjamn/recast/issues?q=is%3Aissue+is%3Aopen+parenthesis I think it's by large a Recast architectural flaw.